run_kraken.py

```diff
@@ -1,15 +1,111 @@
+        if not cerberus_status:
+            logging.error("Received a no-go signal from Cerberus, looks like "
+                          "the cluster is unhealthy. Please check the Cerberus "
+                          "report for more details. Test failed.")
+            sys.exit(1)
+        else:
+            logging.info("Received a go signal from Ceberus, the cluster is healthy. "
+                         "Test passed.")
+
+
+# Get the node scenarios object of specfied cloud type
+def get_node_scenario_object(node_scenario):
+    if node_scenario['cloud_type'] == 'aws':
+        return aws_node_scenarios()
+
+
+# Inject the specified node scenario
+def inject_node_scenario(action, node_scenario, node_scenario_object):
+    # Get the node scenario configurations
+    instance_kill_count = node_scenario.get("instance_kill_count", 1)
+    node_name = node_scenario.get("node_name", "")
+    label_selector = node_scenario.get("label_selector", "")
+    timeout = node_scenario.get("timeout", 120)
+
+    # Get the node to apply the scenario
+    node = nodeaction.get_node(node_name, label_selector)
+
+    if action == "node_start_scenario":
+        node_scenario_object.node_start_scenario(instance_kill_count, node, timeout)
+    elif action == "node_stop_scenario":
+        node_scenario_object.node_stop_scenario(instance_kill_count, node, timeout)
+    elif action == "node_stop_start_scenario":
+        node_scenario_object.node_stop_start_scenario(instance_kill_count, node, timeout)
+    elif action == "node_termination_scenario":
+        node_scenario_object.node_termination_scenario(instance_kill_count, node, timeout)
+    elif action == "node_reboot_scenario":
+        node_scenario_object.node_reboot_scenario(instance_kill_count, node, timeout)
+    elif action == "stop_kubelet_scenario":
+        node_scenario_object.stop_kubelet_scenario(instance_kill_count, node, timeout)
+    elif action == "stop_start_kubelet_scenario":
+        node_scenario_object.stop_start_kubelet_scenario(instance_kill_count, node, timeout)
+    elif action == "node_crash_scenario":
+        node_scenario_object.node_crash_scenario(instance_kill_count, node, timeout)
+
+
+# Inject the cluster shut down scenario
+def cluster_shut_down(shut_down_config, cerberus_enabled, config):
+    runs = shut_down_config["runs"]
+    shut_down_duration = shut_down_config["shut_down_duration"]
+    cloud_type = shut_down_config["cloud_type"]
+    if cloud_type == "aws":
+        cloud_object = AWS()
+
+    nodes = set(kubecli.list_nodes())
+    node_id = {}
+    for node in nodes:
+        node_id[node] = cloud_object.get_instance_id(node)
+
+    for _ in range(runs):
+        logging.info("Starting cluster_shut_down scenario injection")
+        for node in nodes:
+            cloud_object.stop_instances(node_id[node])
+        logging.info("Waiting for 250s to shut down all the nodes")
+        time.sleep(250)
```

Comment:
Are we able to have the user set this in their config or use 250 as a default? Is there a specific reason we chose 250 seconds here?

Reply:
There's no specific reason, it took around 2 mins on 10 node cluster. So I chose 250 seconds to accommodate clusters of bigger sizes. [start_instance](https://github.com/openshift-scale/kraken/pull/25/files#diff-cfec6e969439720ae75ffa7156ac298bR98) function can be called on a node only when it is in stopped state else it throws error. However I have added a [try except](https://github.com/openshift-scale/kraken/pull/25/files#diff-cfec6e969439720ae75ffa7156ac298bR97) condition such that we sleep for 10 additional seconds when a node isn't in stopped state even after 250 seconds.